### PR TITLE
Add csv route for creating certificates

### PR DESCRIPTION
**Pull request by junron**
[task](https://codein.withgoogle.com/dashboard/task-instances/6571674569801728/)

Added CSV API view for creating certs.
Only allows organizations to create certs.

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -17,6 +18,33 @@
...
+            file = request.FILES["file"]
+            if(file.content_type!="text/csv"):
+                return Response("Please only upload csv files", status=status.HTTP_400_BAD_REQUEST)
+            fileData = file.read().decode("utf-8")
+            csvData = list(csv.reader(fileData.split("\n")))
+            csvData.pop(0)
+            for row in csvData:
+                # First column is issued_to
+                # Second column is issued_for
+                if(len(row)!=2):
```
Comment (jatinkatyal13) [inline api/views.py]: May be in future we'll have more data in certificate model we don't want this static! Isn't there a way to find number of columns that we need?

File: api/views.py [hunk 2]
```diff
@@ -17,6 +18,33 @@
...
+                    continue
+                print(f"{request.user.organization.pk} issued a certificate to '{row[0]}' for '{row[1]}'")
```
Comment (jatinkatyal13) [inline api/views.py]: Remove the print statement! It is the best way to debug 😜 but not logging!

File: api/views.py [hunk 3]
```diff
@@ -17,6 +18,33 @@
...
+                data = {
+                    "student":User.objects.get(username=row[0]),
+                    "issued_for":row[1],
+                    "issuing_organization":request.user.organization.pk
+                }
+                serializer = self.serializer_class(data=data)
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+            return Response("Certificates created succesfully.", status=status.HTTP_201_CREATED, headers=headers)
```
Comment (jatinkatyal13) [inline api/views.py]: We should return all the generated certificates as response! @aashutoshrathi what say?

File: api/views.py (hunk 1 above)
Comment (junron) [reply api/views.py]: Hmm perhaps a list that stores the columns?

File: api/views.py (hunk 2 above)
Comment (junron) [reply api/views.py]: Removed

File: api/views.py (hunk 1 above)
Comment (junron) [reply api/views.py]: @jatinkatyal13 checkout the changes I've made

File: api/views.py (hunk 3 above)
Comment (jatinkatyal13) [reply api/views.py]: @junron lastly this !

Comment (junron) [reply api/views.py]: Added

Comment (junron) [reply api/views.py]: @jatinkatyal13 have a look

File: api/views.py [hunk 4]
```diff
@@ -16,7 +17,34 @@
...
+            if(file.content_type!="text/csv"):
+                return Response("Please only upload csv files", status=status.HTTP_400_BAD_REQUEST)
+            fileData = file.read().decode("utf-8")
+            csvData = list(csv.reader(fileData.split("\n")))
+            csvData.pop(0)
+            certificates = []
+            for row in csvData:
+                # First column is issued_to
+                # Second column is issued_for
+                if(len(row)!=len(model.getFields())):
```
Comment (jatinkatyal13) [inline api/views.py]: It should be:
`
if(len(row)!=len(self.model.getFields())):
`

File: api/views.py [hunk 5]
```diff
@@ -16,7 +17,34 @@
...
+                    continue
+                data = {
+                    "student":User.objects.get(username=row[0]),
```
Comment (jatinkatyal13) [inline api/views.py]: certificate serializer expects a student object! This gives an error! It should be
`
"student":models.Student.objects.get(username=row[0]).pk,
`

**Comment by jatinkatyal13** [comment]
Also remove the headers from the Response returned as it gives undefined error!

File: api/views.py (hunk 5 above)
Comment (junron) [reply api/views.py]: Oops, fixed that and squashed commits

**Comment by junron** [comment]
Fixed highlighted errors

**Comment by junron** [comment]
Done

**Pull request merged**